src/relay/pass/partial_eval.cc

```diff
@@ -648,8 +648,11 @@ class PartialEvaluator : public ExprFunctor<PStatic(const Expr& e, LetList* ll)>
           if (time_map_.count(fid) == 0) {
             return recurse();
           } else {
-            // We check to see that at least one argument decrease
-            // with respect to all previous invocation.
+            /* We check to see that at least one argument decrease
+             * with respect to all previous invocation.
+             * The depth of the recursion is bounded by
+             * the sum of the time of all argument at the first call.
```

Comment:
I think it's product in stead of sum?

Reply:
it is sum. everytime the sum must decrease by at least 1.

Reply:
yes you are right. @MarisaKirisame